backend/chat/moderation/raid-message-checker.js

```diff
@@ -0,0 +1,90 @@
+"use strict";
+
+const logger = require("../../logwrapper");
+
+let messageCache = [];
+let chatCacheLimit = 50;
+
+let raidMessage = "";
+let checkerEnabled = false;
+let settings = {
+    shouldBan: false,
+    shouldBlock: false
+};
+
+function handleRaider(message) {
+    const chat = require("../twitch-chat");
+
+    if (settings.shouldBan) {
+        chat.ban(message.username, "");
+    }
+
+    if (settings.shouldBlock) {
+        chat.block(message.userId, "");
+    }
+}
+
+/**
+ *
+ * @param {import("../chat-helpers").FirebotChatMessage} firebotChatMessage
+ */
+function sendMessageToCache(firebotChatMessage) {
+    if (messageCache.length >= chatCacheLimit) {
+        messageCache.shift();
+    }
+
+    if (firebotChatMessage.rawText.length > 10) {
+        firebotChatMessage.rawText = firebotChatMessage.rawText.substr(10);
+    }
+
+    messageCache.push(firebotChatMessage);
+
+    if (firebotChatMessage && checkerEnabled && firebotChatMessage.rawText === raidMessage) {
+        handleRaider(firebotChatMessage);
+    }
+}
+
+function getRaidMessage() {
+    let rawMessages = messageCache.map(message => message.rawText);
+    let raidMessages = rawMessages.reduce(function (allMessages, message) {
+        if (message in allMessages) {
+            allMessages[message]++;
+        } else {
+            allMessages[message] = 1;
+        }
+
+        return allMessages;
+    }, {});
```

Comment:
LOVE the use of `.reduce`. Couple relatively small stylistic things I would tweak however:
1. Use a fat arrow function to be consistent with everything else in `getRaidMessage` 
```js
rawMessages.reduce((allMessages, message) => {
```

2. Instead of using `if(message in allMessages)`, use `if(allMessages[message] != null)`. 
The `in` operator only checks for the existence of a property, it doesn't check the value for that property. Showcase of issue:
```js  
const foo = { bar: undefined };
console.log("bar" in foo); // outputs: true
```
> Note it returns true even though the property is undefined which wouldn't be good if we were trying to increment a number by 1. This shortcoming wouldn't be a problem where you are currently using it, but it's just a best practice to avoid using that all together so you never have to work about its gotcha's. 


3. Speaking of gotchas, it's typically best to use `+= 1` instead of `++` to increment a number by 1. This is because ++ has really weird behaviors associated to it that can make unexpected results happen if you aren't careful.

Gotcha showcase:
```js
function incrementTest() {
    let num = 1;
    return num++;
}
console.log(incrementTest()); // Output: 1
```
> Note it returned 1 instead of 2. This is because ++ doesn't increment until AFTER the function returns. Weird right? I honestly don't know why it behaves this way. In any case, `+= 1` doesn't have this drawback and has the added benefit of being less ambiguous as well.

Reply:
Woahh, good points!! Updated the PR to use all of your suggestions, and I also fixed the const/let thing. Thank you!!